presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -692,34 +689,101 @@ protected ExtractionResult visitNullLiteral(NullLiteral node, Boolean complement
 
         Type leftType = expressionTypes.get(comparison.getLeft());
         Type rightType = expressionTypes.get(comparison.getRight());
-
         Preconditions.checkArgument(leftType.equals(rightType), "left and right type do not match in comparison expression (%s)", comparison);
 
-        if (left instanceof SymbolReference && !(right instanceof Expression)) {
-            return Optional.of(new NormalizedSimpleComparison((SymbolReference) left, comparison.getType(), new NullableValue(rightType, right)));
+        if (left instanceof Expression && right instanceof Expression) {
+            return Optional.empty();
+        }
+
+        if (!(left instanceof Expression) && !(right instanceof Expression)) {
```

Comment:
isn't that technically illegal argument too? comparison should fold to constant if both sides are evaluated